src/platform/plugins/shared/dashboard/public/dashboard_api/unsaved_changes_manager.ts

```diff
@@ -82,9 +94,10 @@ export function initializeUnsavedChangesManager({
     settingsManager.internalApi.startComparing$(lastSavedState$),
     unifiedSearchManager.internalApi.startComparing$(lastSavedState$),
     layoutManager.internalApi.startComparing$(lastSavedState$),
+    projectRoutingManager?.internalApi.startComparing$(lastSavedState$) ?? of({}),
   ]).pipe(
-    map(([settings, unifiedSearch, panels]) => {
-      return { ...settings, ...unifiedSearch, ...panels };
+    map(([settings, unifiedSearch, panels, projectRouting]) => {
+      return { ...settings, ...unifiedSearch, ...panels, ...projectRouting };
```

Comment:
When testing `3. Save With Custom Project Routing` the only way to open the Save Modal is by clicking on the `Save as`.
This seems to be caused by the fact that the unsaved_changes_manager is not picking up the change of project routing as an actual change in the dashboard.
Is that a desired behaviour?